code/modules/events/minor/special_order.dm

```diff
@@ -0,0 +1,401 @@
+		for(var/obj/item/storage/S in contents)
+			contents |= S.get_all_contents()
+
+		. = TRUE
+
+		for(var/obj/item/reagent_containers/F in contents)
+			for(var/I in food_count)
+				if(F.type == I)
+					food_count[I] -= max(1, F.amount)
+		for(var/I in food_count)
+			if(food_count[I] > 0)
+				return FALSE
+		return TRUE
+
+	get_shopping_list()
+		. = "<ul>"
+		for(var/f_type in src.food_order)
+			var/obj/item/reagent_containers/F = f_type
+			if(src.food_order[f_type])
+				. += "<li>([src.food_order[f_type]]) [initial(F.name)]</li>"
+			else
+				. += "<li>[initial(F.name)]</li>"
+		. += "</ul>"
+
+	breakfast_order
+		name = "Breakfast Order"
+
+		New()
+			if(!breakfast)
+				breakfast = list()
+				for(var/food_type in concrete_typesof(/obj/item/reagent_containers/food/snacks))
+					var/obj/item/reagent_containers/food/snacks/F = food_type
+					if(initial(F.meal_time_flags) & MEAL_TIME_BREAKFAST)
+						breakfast += food_type
+			for(var/i in 1 to rand(3,6))
+				src.food_order[pick(breakfast)] = pick(60; 1, 30; 2, 10; 3)
+			..()
+
+	lunch_order
+		name = "Lunch Order"
+
+		New()
+			if(!lunch)
+				lunch = list()
+				for(var/food_type in concrete_typesof(/obj/item/reagent_containers/food/snacks))
+					var/obj/item/reagent_containers/food/snacks/F = food_type
+					if(initial(F.meal_time_flags) & MEAL_TIME_LUNCH)
+						lunch += food_type
+			for(var/i in 1 to rand(3,6))
+				src.food_order[pick(lunch)] = pick(60; 1, 30; 2)
+			..()
+
+	dinner_order
+		name = "Dinner Order"
+
+		New()
+			if(!dinner)
+				dinner = list()
+				for(var/food_type in concrete_typesof(/obj/item/reagent_containers/food/snacks))
+					var/obj/item/reagent_containers/food/snacks/F = food_type
+					if(initial(F.meal_time_flags) & MEAL_TIME_DINNER)
+						dinner += food_type
+			for(var/i in 1 to rand(3,6))
+				src.food_order[pick(dinner)] = pick(60; 1, 30; 2)
+			..()
+
+	snack_order
+		name = "Snack Order"
+
+		New()
+			if(!snacks)
+				snacks = list()
+				for(var/food_type in concrete_typesof(/obj/item/reagent_containers/food/snacks))
+					var/obj/item/reagent_containers/food/snacks/F = food_type
+					if(initial(F.meal_time_flags) & MEAL_TIME_SNACK)
+						snacks += food_type
+			for(var/i in 1 to rand(3,6))
+				src.food_order[pick(snacks)] = pick(40; 1, 40; 2, 20; 3)
+			..()
+
+/datum/special_order/reagents/blood
+	name = "Blood Request"
+	requisition = new /obj/item/paper/requisition/blood
+	price = 9000
+	reagents_list = list("blood"=1000)
+
+/datum/special_order/surgery/organ_swap
+	name = "Organ Swap"
+	weight = 25
+	price = 5000
+	C = new /obj/storage/crate/wooden
+	requisition = new /obj/item/paper/requisition/surgery/organ_swap
+	var/mob/living/carbon/human/target
+	var/target_organs = list()
+
+	New()
+		..()
+		var/possible_targets = list("brain", "left_eye", "right_eye", "heart", "left_lung", "right_lung", "butt", "left_kidney", "right_kidney", "liver", "stomach", "intestines", "spleen", "pancreas", "appendix")
```

Comment:
maybe exclude brain and stipulate that the dude has to be alive? (or reduced payout if dead)

Reply:
Grading is a possible future improvement for sure.

The premise of the order is practice of limb replacement the twist is that the subject is alive.  Accompanying message  is intentionally vague regarding the current state of the "individual" and any information related to them is to be ignored and is all part of the "training exercise".

If you feel the twist/flavor is in poor taste I can update it.

Reply:
I was just thinking along the lines of surgeon simulator where you gotta keep the "patient" alive:tm:
No strong feelings either way

Reply:
I think that would be a great variation where a "patient" is provided needing organ repair/replacement or a malady that requires specific types of intervention.  "Local Medbay Full - Overflow Patient"